src/map.js

```diff
@@ -77,6 +78,9 @@ var LMap = L.Map.extend({
             this.addControl(this.shareControl);
         }
 
+        this._mapboxLogoControl = mapboxLogoControl({urlOrMapID: _});
```

Comment:
Instead of `{urlOrMapID: _}`, which is no longer used, pass `this.options.mapboxLogoControl` here, which will allow:

```
L.mapbox.map(map, mapid, {mapboxLogoControl: {position: 'topright'}});
```